cli/azd/cmd/vs_server.go

```diff
@@ -63,18 +64,24 @@ func (s *vsServerAction) Run(ctx context.Context) (*actions.ActionResult, error)
 		panic(err)
 	}
 
-	res, err := json.Marshal(struct {
-		Port int `json:"port"`
-		Pid  int `json:"pid"`
-	}{
-		Port: listener.Addr().(*net.TCPAddr).Port,
-		Pid:  os.Getpid(),
-	})
+	var versionRes contracts.VersionResult
+	versionSpec := internal.VersionInfo()
+
+	versionRes.Azd.Commit = versionSpec.Commit
+	versionRes.Azd.Version = versionSpec.Version.String()
+
+	res := contracts.VsServerResult{
+		Port:          listener.Addr().(*net.TCPAddr).Port,
```

Comment:
this cast looks scary :P

Reply:
We (go devs) prefer living on the edge... But I think it's verifiably guaranteed with `net.Listen("tcp")`, just not represented strongly in type :)